### ISSUE TITLE
Add Wordpress 

### ISSUE THREAD

**Issue by robgates** [GhostWriters/DockSTARTer#350]
Any chance of adding something like wordpress to the list of applications ?

**Comment by nemchik** [comment]
Adding notes:
https://hub.docker.com/_/wordpress/

**Comment by Tharic99** [comment]
Following up on this one as I'd love to see WP added to the list of apps as well!

**Comment by robgates** [comment]
I managed to get it working and have copied the steps below


  1.  Use DS to install MariaDB and phpMyadmin
  2.  Create a database via phpMyAdmin to create a database (named wordpress, created a database id (wordpress) and set a password for the database id.
  3.  Create an override file ~/.docker/compose/docker-compose.override.yml . I used the as below but changed the password to the password I created for the datbase

services:
               wordpress:
container_name: wordpress
environment:
WORDPRESS_DB_HOST=mariadb
WORDPRESS_DB_USER=wordpress
WORDPRESS_DB_PASSWORD=xxxxxxxxx
WORDPRESS_DB_NAME=wordpress
PGID=${PGID}
PUID=${PUID}
TZ=${TZ}
image: wordpress
ports:
8080:80
restart: unless-stopped
volumes:
/etc/localtime:/etc/localtime:ro
${DOCKERCONFDIR}/wordpress:/config
${DOCKERSHAREDDIR}:/shared
version: “3.4”

then just run sudo ds -c



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by nemchik** [comment]
This is nearly done on my end. Just giving a status update. Should be in tomorrow.

**Comment by nemchik** [comment]
After some thought it does not make sense to run WP in Docker due to the file structure and nature of self updating web apps vs docker managed app files. This is explained in detail on our discord server if anyone feels the need to look into a much more wordy explanation.

**Comment by nemchik** [comment]
Was asked about this recently on discord and figured the point explanation might be easier to find here.

Copied from discord:

> WordPress itself doesn't seem to be a good fit inside a docker container. Web servers do well inside docker, and you can run WordPress from a persistent storage volume with a web server in docker. For example DS includes the letsencrypt container and you can put WordPress in it's config folder and let it run from there. You could also run a separate web server like a standalone nginx or something else if you prefer and run WordPress with on of those.
> 
> The main reason for this is that WordPress doesn't seem to do well being "stateless". WordPress requires persistent storage for its config, and content (uploads, themes, plugins, etc). To make things more complicated, some of the files that need to be persistent are in the same folders as other files that would need to be considered part of the container in order to make updates work the way docker is designed. Long story short, many web application are not built to run inside docker containers.
> 
> Another example is Organizr. We have a docker container for it, but in reality what it does is run nginx and put the Organizr web app in a persistent volume (/config) so the docker container is just nginx and some scripts that setup Organizr in the persistent storage. This isn't a bad approach, but I haven't seen anyone make something like this for WordPress. That being said, if you run a web server container and put WordPress in your persistent config volume the only different is that the container didn't do it for you.
> 
> tl;dr yes the current belief is still that WordPress doesn't fit well inside a container
